### PR TITLE
Implementation of equal_scalar() for raw and complex

### PR DESCRIPTION
**Pull request by romainfrancois**
it's a bit odd to have to define a proxy for raw vectors, but R does not otherwise know how to order a raw vector 🤷‍♂ : 

``` r
order(as.raw(1:3))
#> Error in order(as.raw(1:3)): unimplemented type 'raw' in 'orderVector1'
```

<sup>Created on 2019-07-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

### CODE REVIEW

File: src/equal.c [hunk 1]
```diff
@@ -4,6 +4,8 @@
 static int lgl_equal_scalar(const int* x, const int* y, bool na_equal);
 static int int_equal_scalar(const int* x, const int* y, bool na_equal);
 static int dbl_equal_scalar(const double* x, const double* y, bool na_equal);
+static int raw_equal_scalar(const Rbyte* x, const Rbyte* y, bool na_equal);
+static int cplx_equal_scalar(const Rcomplex* x, const Rcomplex* y, bool na_equal);
```
Comment (lionel-) [inline src/equal.c]: We use the `cpl_` particle in rlang and vctrs.

**Comment by romainfrancois** [comment]
This gives this (on https://github.com/tidyverse/dplyr/pull/4504): 

``` r
library(dplyr, warn.conflicts = FALSE)

df <- tibble(a = 1:3, b = as.raw(1:3))
group_by(df, b)
#> # A tibble: 3 x 2
#> # Groups:   b [3]
#>       a b    
#>   <int> <>   
#> 1     1 01   
#> 2     2 02   
#> 3     3 03
```

<sup>Created on 2019-07-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

File: src/equal.c [hunk 2]
```diff
@@ -105,6 +111,10 @@ static int int_equal_scalar(const int* x, const int* y, bool na_equal) {
     return (xi == NA_INTEGER || yj == NA_INTEGER) ? NA_LOGICAL : xi == yj;
   }
 }
+static int raw_equal_scalar(const Rbyte* x, const Rbyte* y, bool na_equal) {
+  // raw vectors have no notion of missing value
```
Comment (lionel-) [inline src/equal.c]: Comments should start with capital and no final stop, e.g. https://style.tidyverse.org/functions.html#comments-1

File: R/type-raw.R [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+#' @export
+vec_proxy_compare.raw <- function(x, ...) {
```
Comment (lionel-) [inline R/type-raw.R]: Could you also add tests for this please?

**Comment by lionel-** [comment]
Thanks!

**Pull request merged**